wemake_python_styleguide/visitors/ast/complexity/function.py

```diff
@@ -42,14 +42,24 @@ class _ComplexityCounter:
     """Helper class to encapsulate logic from the visitor."""
 
     _not_contain_locals: ClassVar[AnyNodes] = (ast.comprehension,)
+    _staticmethod_names: ClassVar[frozenset[str]] = frozenset(('staticmethod',))
 
     def __init__(self) -> None:
         self.metrics = ComplexityMetrics()
 
-    def check_arguments_count(self, node: AnyFunctionDefAndLambda) -> None:
+    def check_arguments_count_any(self, node: AnyFunctionDefAndLambda) -> None:
         """Checks the number of the arguments in a function."""
         self.metrics.arguments[node] = len(functions.get_all_arguments(node))
 
+    def check_arguments_count(self, node: AnyFunctionDef) -> None:
+        """Checks the number of the arguments in a function."""
+        self.check_arguments_count_any(node)
+        parent = get_parent(node)
+        if isinstance(parent, ast.ClassDef) and not self._check_staticmethod(
```

Comment:
Maybe instead we should just ignore first arg named `self` / `cls` / `mc`? We even have `_get_args_without_special_argument` for that.

Reply:
Are you suggesting switching

`wemake_python_styleguide.logic.tree.functions.get_all_arguments`

to

`wemake_python_styleguide.logic.arguments.function_args._get_args_without_special_argument`

around line 51?

Of course I would make the function public and update callers, etc etc. I just want to confirm I understand this right.

Reply:
Yes, this seems like the simpliest solution.

Reply:
I took a look and that doesn't seem to work. The function doesn't grab keyword or variable length arguments. I did refactor a little bit to try and reuse some of that code though.